mla/analysis.py

```diff
@@ -323,47 +323,522 @@ def produce_and_minimize(self, event_model: models.EventModel, # Super class wil
 
         return fit_info
 
+
+class ThreeMLPsAnalysis(Analysis):
+    """Class info...
+    More class info...
+    Attributes:
+        injector (injectors.PsInjector):
+    """
+    def __init__(self,
+                 source: Optional[Dict[str, float]] = None,
+                 injector: Optional[injectors.PsInjector] = None) -> None:
+        """Function info...
+        More function info...
+        Args:
```

Comment:
make sure to call super().__init__() and keep the function signatures the same as the parent